lib/resty/dns/resolver.lua

```diff
@@ -832,6 +839,69 @@ end
 
 
 function _M.query(self, qname, opts, tries)
+    if self.doh then
+        return _M.doh_query(self,qname,opts,tries)
+    end
+
+    return _M.udp_tcp_query(self,qname,opts,tries)
+end
+
+function _M.doh_query(self, qname, opts, tries)
+    local retrans = self.retrans
+    if tries then
+        tries[1] = nil
+    end
+
+    local servers = self.servers
+
+    if #servers == 0 then
+        return nil, "No servers available"
+    end
+
+    local err
+
+    for i = 1, retrans do
+        local idx = i
+
+        if idx > #servers then
+            idx = 1
+        end
+
+        local res
+        local id 
+        
+        if self.doh == 'GET' then
+            res = ngx.location.capture(servers[idx][1] .. b64.encode_base64url(qname))
```

Comment:
We should avoid using `ngx.location.capture` here, a better way is to use cosockets

Reply:
I was trying to use builtin functionality, because I don't think it's a reasonable solution to implement http requests, since library for that exists [lua-resty-http](/ledgetech/lua-resty-http) maybe the best is to pcall(require,"resty.http") and make DoH available only when this library exists.

Reply:
It is not recommended to rely on lua-resty-http module here, perhaps you could follow lua-resty-http to make a simple wrapper around the http client